File: code/jvm/src/main/kotlin/pt/isel/leic/ptgest/http/controllers/workout/model/request/CreateCustomSetRequest.kt
package pt.isel.leic.ptgest.http.controllers.workout.model.request

import pt.isel.leic.ptgest.domain.workout.SetType
import pt.isel.leic.ptgest.domain.workout.model.SetExercise

data class CreateCustomSetRequest(
    val name: String?,
    val notes: String?,
    val type: SetType,
    val setExercises: List<SetExercise>
) {
    init {
        name?.trim()
        notes?.trim()
    }
}

File: code/js/src/views/user/TrainerViews/models/CreateCustomSetRequest.ts
import SetExercise from "./SetExercise.ts";

class CreateCustomSetRequest {
    name: string | null
    notes: string | null
    type: string
    setExercises: SetExercise[]

    constructor(name: string | null, notes: string | null, type: string, setExercises: SetExercise[]){
        this.name = name
        this.notes = notes
        this.type = type
        this.setExercises = setExercises
    }
}
export default CreateCustomSetRequest;
